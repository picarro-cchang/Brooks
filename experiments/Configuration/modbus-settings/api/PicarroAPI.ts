export const PicarroAPI = {
  postData(url: string, data: object) {
    console.log(data);
    return fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    }).then(response => {
      if (!response.ok) {
        throw Error('Network POST request failed');
      }
      return response;
    });
  },

  getRequest(url: string) {
    return fetch(url, {
      method: 'GET',
    }).then(response => {
      if (!response.ok) {
        throw Error('Network GET request failed');
      }
      return response.json();
    });
  },

  getRequestWithHeader(url: string, header: Headers) {
    return fetch(url, {
      headers: header,
      method: 'GET',
    }).then(response => {
      if (!response.ok) {
        throw Error('Network GET request failed');
      }
      return response.json();
    });
  },
};

//export default PicarroAPI;
