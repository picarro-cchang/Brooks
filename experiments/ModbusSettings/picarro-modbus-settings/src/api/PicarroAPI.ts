let PicarroAPI = {
  postData(url: string, data: object) {
    console.log(data);
    return fetch(url, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'text/html',
      },
      body: JSON.stringify(data),
    }).then(response => {
      if (!response.ok) {
        throw Error('Network POST request failed');
      }
      return response;
    });
  },

  getRequest(url: string) {
    return fetch(url, {
      method: 'GET',
    }).then(response => {
      if (!response.ok) {
        throw Error('Network GET request failed');
      }
      return response.json();
    });
  },
};

export default PicarroAPI;
