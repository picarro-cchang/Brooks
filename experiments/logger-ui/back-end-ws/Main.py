from aiohttp import web
import asyncio
import aiohttp_cors
from aiohttp_swagger import setup_swagger

from experiments.common.async_helper import log_async_exception

from LoggerServer import LoggerServer


class Main:

    """
    This class will create a top level server for logger application. It will
    create a top level web application, that will have a logger server as 
    subapp.
    """

    def __init__(self, port=9090, addr="0.0.0.0", identifier="Logger"):
        self.tasks = []
        self.ws_send_queue = asyncio.Queue(100)
        self.port = port
        self.addr = addr
        self.runner = None
        self.identifier = identifier
        self.ws_send_queue = asyncio.Queue(256)

    async def on_startup(self, app):
        print(f"Top Level Server is Starting Up {self.addr}:{self.port}")

    async def on_shutdown(self, app):
        for task in self.tasks:
            task.cancel()
        print("Top level server is shutting down")

    async def on_cleanup(self, app):
        print("TODO: Implement cleanup")
        pass

    async def server_init(self):
        self.app = web.Application()
        self.app.on_startup.append((self.on_startup))
        self.app.on_shutdown.append((self.on_shutdown))
        self.app.on_cleanup.append((self.on_cleanup))

        cors = aiohttp_cors.setup(
            self.app, defaults={"*": aiohttp_cors.ResourceOptions(
                allow_credentials=True,
                expose_headers="*",
                allow_headers="*",
            )})

        api_ws_server = LoggerServer()
        api_ws_server.app['send_queue'] = self.ws_send_queue
        self.app.add_subapp("/api_ws/", api_ws_server.app)
        setup_swagger(self.app)

        for route in self.app.router.routes():
            cors.add(route)

        self.runner = web.AppRunner(self.app)
        await self.runner.setup()
        site = web.TCPSite(self.runner, host=self.addr, port=self.port)
        await site.start()
        print(f"\n>_ Running on http://{site._host}:{site._port} ...\n")

    async def startup(self):
        self.tasks.append(asyncio.create_task(self.server_init()))

    async def cleanup(self):
        self.tasks.append(asyncio.create_task(self.on_cleanup()))


if __name__ == "__main__":
    service = Main(port=8004)
    loop = asyncio.get_event_loop()
    loop.run_until_complete(asyncio.gather(service.startup()))
    try:
        loop.run_forever()
    except KeyboardInterrupt:
        print("Stop Server Begin...")
    finally:
        loop.run_until_complete(asyncio.gather(service.runner.cleanup()))
        loop.close()
    print("Stop Server Ended")
