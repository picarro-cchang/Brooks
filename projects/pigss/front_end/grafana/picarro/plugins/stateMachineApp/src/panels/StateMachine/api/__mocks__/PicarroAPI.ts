const apiLoc = `http://localhost:8000/controller`;

let PicarroAPI = {
    getRequest(url: string) {
        switch(url) {
            case apiLoc+'/uistatus': {
                const data = {
                    "standby": "ACTIVE",
                    "identify": "READY",
                    "loop_plan": "READY",
                    "run": "READY",
                    "plan": "READY",
                    "plan_run": "READY",
                    "plan_loop": "READY",
                    "reference": "READY",
                    "run_plan": "READY",
                    "edit": "READY",
                    "clean": {
                      "1": "READY",
                      "3": "READY",
                      "4": "READY"
                    },
                    "bank": {
                      "1": "READY",
                      "3": "READY",
                      "4": "READY"
                    },
                    "channel": {
                      "1": {
                        "1": "DISABLED",
                        "2": "AVAILABLE",
                        "3": "DISABLED",
                        "4": "AVAILABLE",
                        "5": "DISABLED",
                        "6": "AVAILABLE",
                        "7": "DISABLED",
                        "8": "AVAILABLE"
                      },
                      "3": {
                        "1": "DISABLED",
                        "2": "AVAILABLE",
                        "3": "DISABLED",
                        "4": "AVAILABLE",
                        "5": "DISABLED",
                        "6": "AVAILABLE",
                        "7": "DISABLED",
                        "8": "AVAILABLE"
                      },
                      "4": {
                        "1": "DISABLED",
                        "2": "AVAILABLE",
                        "3": "DISABLED",
                        "4": "AVAILABLE",
                        "5": "DISABLED",
                        "6": "AVAILABLE",
                        "7": "DISABLED",
                        "8": "AVAILABLE"
                      }
                    }
                }
                return Promise.resolve(new Response(JSON.stringify(data)))
            }
            case apiLoc+'/plan': {
                const plan = {
                    "max_steps": 32,
                    "panel_to_show": 0,
                    "current_step": 11,
                    "looping": false,
                    "focus": {
                      "row": 12,
                      "column": 1
                    },
                    "last_step": 11,
                    "steps": {
                      "1": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 2
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 0
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "2": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 32
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 0
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "3": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 8
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 0
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "4": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 128
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 0
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "5": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 32
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 0
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "6": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 8
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 0
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "7": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 128
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 0
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "8": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 2
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "9": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 32
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "10": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 8
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      },
                      "11": {
                        "banks": {
                          "1": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "3": {
                            "clean": 0,
                            "chan_mask": 0
                          },
                          "4": {
                            "clean": 0,
                            "chan_mask": 128
                          }
                        },
                        "reference": 0,
                        "duration": 20
                      }
                    },
                    "num_plan_files": 2,
                    "plan_files": {
                      "1": "__default__",
                      "2": "tr"
                    },
                    "plan_filename": "__default__",
                    "bank_names": {
                      "1": {
                        "name": "Bank 1",
                        "channels": {
                          "1": "Ch. 1",
                          "2": "Ch. 2",
                          "3": "Ch. 3",
                          "4": "Ch. 4",
                          "5": "Ch. 5",
                          "6": "Ch. 6",
                          "7": "Ch. 7",
                          "8": "Ch. 8"
                        }
                      },
                      "2": {
                        "name": "Bank 2",
                        "channels": {
                          "1": "Ch. 1",
                          "2": "Ch. 2",
                          "3": "Ch. 3",
                          "4": "Ch. 4",
                          "5": "Ch. 5",
                          "6": "Ch. 6",
                          "7": "Ch. 7",
                          "8": "Ch. 8"
                        }
                      },
                      "3": {
                        "name": "Bank 3",
                        "channels": {
                          "1": "Ch. 1",
                          "2": "Ch. 2",
                          "3": "Ch. 3",
                          "4": "Ch. 4",
                          "5": "Ch. 5",
                          "6": "Ch. 6",
                          "7": "Ch. 7",
                          "8": "Ch. 8"
                        }
                      },
                      "4": {
                        "name": "Bank 4",
                        "channels": {
                          "1": "Ch. 1",
                          "2": "Ch. 2",
                          "3": "Ch. 3",
                          "4": "Ch. 4",
                          "5": "Ch. 5",
                          "6": "Ch. 6",
                          "7": "Ch. 7",
                          "8": "Ch. 8"
                        }
                      }
                    },
                    "available_ports": {
                      "1": 170,
                      "2": 0,
                      "3": 170,
                      "4": 170
                    }
                }
                return Promise.resolve(new Response(JSON.stringify(plan)))
            }
            case apiLoc+'/modal_info': {
                const info = {
                    "show": false,
                    "html": "<h2 class='test'>Confirm Loop Plan</h2><p>Loop plan starting at step 1</p>",
                    "num_buttons": 2,
                    "buttons": {
                      "1": {
                        "caption": "OK",
                        "className": "btn btn-success btn-large",
                        "response": "modal_ok"
                      },
                      "2": {
                        "caption": "Cancel",
                        "className": "btn btn-danger btn-large",
                        "response": "modal_close"
                      }
                    }
                }
                return Promise.resolve(new Response(JSON.stringify(info)))
            }
        }
    //   return fetch(url, {
    //     method: "GET"
    //   }).then(response => {
    //     if (!response.ok) {
    //       throw Error("Network GET request failed");
    //     }
    //     return response;
    //   });
    }
  };
  
  export default PicarroAPI;
  