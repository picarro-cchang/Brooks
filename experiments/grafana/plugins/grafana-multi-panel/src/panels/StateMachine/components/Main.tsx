import React, { Component, PureComponent, useImperativeHandle } from 'react';
import PicarroAPI from './../api/PicarroAPI';
import BankPanel from './BankPanel';
import CommandPanel from './CommandPanel';
import PlanPanel from './PlanPanel';
import PlanLoadPanel from './PlanLoadPanel';
//import * as styles from './module.css';

import PlanSavePanel from './PlanSavePanel';
import deepmerge from 'deepmerge';
import Modal from 'react-responsive-modal';
import {ButtonInfo, ModalInfo, PlanFocus, PlanPanelTypes, PlanStep} from './../types';


const socketURL = `ws://${window.location.hostname}:8000/ws`
export class Main extends Component<any, any> {
  state = {
    initialized: false,
    modal_info: {
      show: false,
      html: "",
      num_buttons: 0,
      buttons: {}
    },
    uistatus: {},
    plan: {
      max_steps: 10,
      panel_to_show: 0,
      focus: { row: 0, column: 0 },
      last_step: 0,
      steps: {},
      num_plan_files: 0,
      plan_files: {},
      plan_filename: ""
    }
  };

  ws = new WebSocket(socketURL);
  componentDidMount() {
    this.getDataViaApi();

    this.ws.onopen = () => {
      // on connecting, do nothing but log it to the console
      console.log('web socket connected');
    }

    this.ws.onmessage = evt => {
      // on receiving a message, add it to the list of messages
      this.handleData(evt.data);
    }

    this.ws.onclose = () => {
      alert('web socket disconnected');
      this.ws = new WebSocket(socketURL);
      this.getDataViaApi();
    }
  }

  getDataViaApi = () => {
    let p1 = PicarroAPI.getRequest(`http://${window.location.hostname}:8000/uistatus`).then(
        response => {
          response.json().then(obj => {
            // this.refWebSocket.sendMessage("message via websocket");
            this.setState(deepmerge(this.state, { uistatus: obj }));
          })
        }
    );
    let p2 = PicarroAPI.getRequest(`http://${window.location.hostname}:8000/plan`).then(
        response => {
          response.json().then(obj => {
            // this.refWebSocket.sendMessage("message via websocket");
            this.setState(deepmerge(this.state, { plan: obj }));
          })
        }
    );
    let p3 = PicarroAPI.getRequest(`http://${window.location.hostname}:8000/modal_info`).then(
        response => {
          response.json().then(obj => {
            // this.refWebSocket.sendMessage("message via websocket");
            this.setState(deepmerge(this.state, { modal_info: obj }));
          })
        }
    );
    Promise.all([p1, p2, p3]).then(() => {
      this.setState(deepmerge(this.state, { initialized: true }));
      console.log("State after getting uistatus, plan and modal info", this.state);
    })
  }

  setFocus(row: number, column: number) {
    this.setState(deepmerge(this.state, { plan: { focus: { row, column } } }));
  }

  handleData(data: any) {
    console.log("Receiving from websocket", data);
    const o = JSON.parse(data);
    if (this.state.initialized) {
      if ("uistatus" in o) {
        const uistatus = deepmerge(this.state.uistatus, o.uistatus);
        this.setState({ uistatus });
      }
      else if ("plan" in o) {
        const plan = deepmerge(this.state.plan, o.plan);
        this.setState({ plan });
      }
      else if ("modal_info" in o) {
        const modal_info = deepmerge(this.state.modal_info, o.modal_info);
        this.setState({ modal_info });
      }
    }
  }

  ws_sender = (o: object) => {
    console.log("Sending to websocket:", o);
    this.ws.send(JSON.stringify(o));
  };

  render() {
    let left_panel;
    switch (this.state.plan.panel_to_show) {
      case PlanPanelTypes.NONE:
        left_panel = (<CommandPanel uistatus={this.state.uistatus} ws_sender={this.ws_sender} />);
        break;
      case PlanPanelTypes.PLAN:
        left_panel = (
            <PlanPanel uistatus={this.state.uistatus} plan={this.state.plan}
                       setFocus={(row, column) => this.setFocus(row, column)}
                       ws_sender={this.ws_sender} />
        )
        break;
      case PlanPanelTypes.LOAD:
        left_panel = (
            <PlanLoadPanel plan={this.state.plan}
                           ws_sender={this.ws_sender} />
        )
        break;
      case PlanPanelTypes.SAVE:
        left_panel = (
            <PlanSavePanel plan={this.state.plan}
                           ws_sender={this.ws_sender} />
        )
        break;
    }

    let modalButtons = [];
    for (let i = 1; i <= this.state.modal_info.num_buttons; i++) {
      const modal_info = this.state.modal_info as ModalInfo;
      modalButtons.push(
          <button className={modal_info.buttons[i].className}
                  style={{ margin: "10px"}}
                  onClick={() => this.ws_sender({ element: modal_info.buttons[i].response })}>
            {modal_info.buttons[i].caption}
          </button>
      )
    }
//TODO: Make the banks only show if they are available/on/exist
    return (
        <div style={{ textAlign: 'center' }}>
          <div className="container-fluid">
            <div className="row justify-content-md-center">
              <div className="col-sm-3">
                {left_panel}
              </div>
              <div className="col-sm-2">
                <BankPanel bank={1} uistatus={this.state.uistatus} ws_sender={this.ws_sender} />
              </div>
              <div className="col-sm-2">
                <BankPanel bank={2} uistatus={this.state.uistatus} ws_sender={this.ws_sender} />
              </div>
              <div className="col-sm-2">
                <BankPanel bank={3} uistatus={this.state.uistatus} ws_sender={this.ws_sender} />
              </div>
              <div className="col-sm-2">
                <BankPanel bank={4} uistatus={this.state.uistatus} ws_sender={this.ws_sender} />
              </div>
            </div>
          </div>
          <Modal open={this.state.modal_info.show} onClose={() => this.ws_sender({ element: "modal_close" })} center>
            <div style={{ margin: "20px"}}>
              <div dangerouslySetInnerHTML={{ __html: this.state.modal_info.html }}>
              </div>
            </div>
            <div style={{color: "black"}}>
              {modalButtons}
            </div>
          </Modal>
        </div>
    );
  }
}

//export default Main;






















// import React, { PureComponent } from "react";
// import { PanelProps, getTheme } from "@grafana/ui";
// import { Options } from "../types";
//
// interface Props extends PanelProps<Options> {}
//
// interface State {
//   error: string;
//   imageUrl: string;
// }
//
// export class Main extends PureComponent<Props, State> {
//
//   constructor(props) {
//     super(props);
//
//     this.state = {
//       error: null,
//       imageUrl: null
//     };
//   }
//
//   updateImage = () => {
//     if (!this.props.options.imageUrl) {
//       return;
//     }
//
//     fetch(this.props.options.imageUrl, {
//       method: "head",
//       mode: "cors"
//     })
//       .then(response => {
//         if (response.status >= 400) {
//           throw Error(response.statusText);
//         }
//
//         const url = this.props.options.imageUrl;
//         this.setState({imageUrl: url, error: null });
//
//       })
//       .catch((error: Error) =>
//         this.setState({
//           imageUrl: null,
//           error: error.message
//         })
//       );
//   };
//
//
//   componentDidMount() {
//     this.updateImage();
//   }
//
//
//   render() {
//     if (!this.state.imageUrl) {
//       return null;
//     }
//
//     if (this.state.error) {
//       const theme = getTheme();
//       return (
//         <strong style={{ color: theme.colors.critical }}>
//           Could not load image: {this.state.error}
//         </strong>
//       );
//     }
//
//     return (
//       <img Hello-Editor={this.state.imageUrl} width={this.props.width} height="auto" />
//     );
//   }
// }
