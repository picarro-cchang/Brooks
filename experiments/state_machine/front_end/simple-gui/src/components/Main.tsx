import React, {Component, PureComponent} from 'react';
import PicarroAPI from '../api/PicarroAPI';
import BankPanel from './BankPanel';
import CommandPanel from './CommandPanel';

interface MainState {
    uistatus: object;
}

  
class Main extends Component<any, any> {
    state = { uistatus:{} };

    onButtonClick(e: any) {
        PicarroAPI.getRequest('http://localhost:8000/uistatus').then(
            response => {
                response.json().then(obj => {
                    this.setState({...{uistatus: obj}})
                })
            }
        );
    }

    render() {
        return (
            <div>
                <button
                    onClick={e => this.onButtonClick(e)}
                    className="btn btn-primary btn-lg">
                    Get Status
                </button>
                <div className="wrapper">
                    <CommandPanel uistatus={this.state.uistatus} />
                    <BankPanel bank={1} uistatus={this.state.uistatus} />
                    <BankPanel bank={2} uistatus={this.state.uistatus} />
                    <BankPanel bank={3} uistatus={this.state.uistatus} />
                    <BankPanel bank={4} uistatus={this.state.uistatus} />
                </div>
            </div>
        );
    }
}

export default Main;