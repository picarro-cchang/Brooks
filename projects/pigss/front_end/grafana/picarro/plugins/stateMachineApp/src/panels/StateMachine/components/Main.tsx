import React, { Component } from "react";
import PicarroAPI from "../api/PicarroAPI";
import deepmerge from "deepmerge";
import { notifyError } from "../utils/Notifications";
import { toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { PlanService } from "../api/PlanService";
import PlanLayout from "./Plan/PlanLayout";
import RunLayout from "./Run/RunLayout";

const REFRESH_INTERVAL = 5;
const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://${apiLoc}/ws`;
export class Main extends React.Component<any, any> {
  state = {
    initialized: false,
    modal_info: {
      show: false,
      html: "",
      num_buttons: 0,
      buttons: {}
    },
    uistatus: {},
    plan: {
      max_steps: 32,
      panel_to_show: 0,
      current_step: 1,
      focus: { row: 1, column: 1 },
      last_step: 0,
      steps: {},
      num_plan_files: 0,
      plan_files: {},
      plan_filename: "",
      bank_names: {
        1: {
          name: "",
          channels: {
            1: "Port 1",
            2: "Port 2",
            3: "Port 3",
            4: "Port 4",
            5: "Port 5",
            6: "Port 6",
            7: "Port 7",
            8: "Port 8"
          }
        },
        2: {
          name: "",
          channels: {
            1: "Port 9",
            2: "Port 10",
            3: "Port 11",
            4: "Port 12",
            5: "Port 13",
            6: "Port 14",
            7: "Port 15",
            8: "Port 16"
          }
        },
        3: {
          name: "",
          channels: {
            1: "Port 17",
            2: "Port 18",
            3: "Port 19",
            4: "Port 20",
            5: "Port 21",
            6: "Port 22",
            7: "Port 23",
            8: "Port 24"
          }
        },
        4: {
          name: "",
          channels: {
            1: "Port 25",
            2: "Port 26",
            3: "Port 27",
            4: "Port 28",
            5: "Port 29",
            6: "Port 30",
            7: "Port 31",
            8: "Port 32"
          }
        }
      }
    },
    options: {
      panel_to_show: 0
    },
    isPlan: false,
    isChanged: false,
    bankAdd: {},
    isLoaded: false,
    fileNames: [],
    isPlanning: false
  };
  constructor(props) {
    super(props);
    this.getPlanFileNames = this.getPlanFileNames.bind(this);
    this.isPlanning = this.isPlanning.bind(this);
    this.getLastRunningPlan = this.getLastRunningPlan.bind(this);
  }

  ws = new WebSocket(socketURL);

  attachWSMethods = (ws: WebSocket) => {
    this.ws.onmessage = evt => {
      // on receiving a message, add it to the list of messages
      this.handleData(evt.data);
    };

    this.ws.onclose = event => {
      if (event.code !== 1000) {
        this.setupWSComm();
      }
    };
  };

  setupWSComm = () => {
    if (this.ws.CLOSED || this.ws.readyState === 0) {
      setTimeout(() => {
        toast.dismiss();
        notifyError("Web Socket Disconnected. Retrying to connect the server.");
        this.ws = new WebSocket(socketURL);
        this.attachWSMethods(this.ws);
        this.getDataViaApi();
      }, REFRESH_INTERVAL * 1000);
    }
  };

  componentDidMount() {
    this.getDataViaApi();
    this.getPlanFileNames();
    this.getLastRunningPlan();
    this.attachWSMethods(this.ws);
  }

  componentWillUnmount() {
    console.log("Component will unmount");
    if (this.ws && this.ws.readyState === 1) {
      this.ws.send("CLOSE");
      this.ws.close(1000, "Client Initited Connection Termination");
    }
  }

  getDataViaApi = () => {
    const uiStatusData = PicarroAPI.getRequest(
      `http://${apiLoc}/uistatus`
    ).then(response => {
      response.json().then(obj => {
        this.setState(deepmerge(this.state, { uistatus: obj }));
      });
    });
    Promise.all([uiStatusData]).then(() => {
      this.setState(deepmerge(this.state, { initialized: true }));
    });
  };


  handleData(data: any) {
    const o = JSON.parse(data);
    console.log("WS ", o)
    if (this.state.initialized) {
      if ("uistatus" in o) {
        const uistatus = deepmerge(this.state.uistatus, o.uistatus);
        this.setState({ uistatus });
      }
      // } else if ("modal_info" in o) {
      //   const modal_info = deepmerge(this.state.modal_info, o.modal_info);
      //   this.setState({ modal_info });
      // }
    }
  }

  ws_sender = (o: object) => {
    this.ws.send(JSON.stringify(o));
  };

  getPlanFileNames() {
    PlanService.getFileNames().then((repsonse: any) => 
    repsonse.json().then(planfiles => {
      this.setState({ fileNames: planfiles['plans'] }, () => console.log(this.state.fileNames));
    })
    )
  }

  planFileNameUpTop() {
    if (this.state.uistatus["plan_loop"] == "ACTIVE" || this.state.uistatus["plan_run"] == "ACTIVE") {
      return true
    } else {
      return false
    }
  }

  isPlanning() {
      this.setState({isPlanning: !this.state.isPlanning})
  }

  getLastRunningPlan() {
    PlanService.getLastRunning().then((response: any) => 
      response.json().then(data => {
        if (data["message"]) {
          console.log("No Plan I suppsoe")
        }          
        else {
          this.setState({plan: data['details']});
        }

      }))
  }

  render() {
  
    return (
        <div> 
            {this.state.isPlanning ? (
                <PlanLayout
                    layoutSwitch={this.isPlanning}
                    fileNames={this.state.fileNames}
                    plan={this.state.plan}
                    uistatus={this.state.uistatus}
                    ws_sender={this.ws_sender}
                    getPlanFileNames={this.getPlanFileNames}
                />
            ):(
                <RunLayout
                    layoutSwitch={this.isPlanning}
                    fileNames={this.state.fileNames}
                    plan={this.state.plan}
                    uistatus={this.state.uistatus}
                    ws_sender={this.ws_sender}
                />
            )}

      
        </div>
    );
  }
}


  //     <Modal
        //   styles={{ overlay: { color: "black" } }}
        //   open={this.state.modal_info.show}
        //   onClose={() => this.ws_sender({ element: "modal_close" })}
        //   center
        // >
        //   <div style={{ margin: "20px" }}>
        //     <div
        //       dangerouslySetInnerHTML={{ __html: this.state.modal_info.html }}
        //     ></div>
        //   </div>
        //   {modalButtons}
        // </Modal>
        // <ToastContainer />
