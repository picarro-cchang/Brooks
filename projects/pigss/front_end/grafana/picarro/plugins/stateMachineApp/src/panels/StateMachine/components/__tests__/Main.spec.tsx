import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import Modal from 'react-responsive-modal';
import WS from 'jest-websocket-mock';
import {Main} from '../Main';
import PicarroAPI from '../../api/PicarroAPI';

const defaultProps = {

};



describe('<Main />', () => {
    // const wrapper = shallow(<Main />);

    it('Snapshot', () => {
        
        // expect(wrapper).toMatchSnapshot();
    });

    it('Check correct left panel is showing', () => {

    });

    it('API test', () => {

    });

    it('handleData', () => {

    });

    it('WS', () => {

    });

    it('updateFileName', () => {

    });

    it('Modal', () => {

    });

    it('Check banks render', () => {

    });
});