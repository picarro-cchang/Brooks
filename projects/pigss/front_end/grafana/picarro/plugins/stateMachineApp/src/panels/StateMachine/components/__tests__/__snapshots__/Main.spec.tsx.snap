// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Snapshot 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    className="container-fluid"
  >
    <div
      className="row justify-content-md-center"
    >
      <div
        className="col-sm-3"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <CommandPanel
          plan={
            Object {
              "bank_names": Object {
                "1": Object {
                  "channels": Object {
                    "1": "",
                    "2": "",
                    "3": "",
                    "4": "",
                    "5": "",
                    "6": "",
                    "7": "",
                    "8": "",
                  },
                  "name": "",
                },
                "2": Object {
                  "channels": Object {
                    "1": "",
                    "2": "",
                    "3": "",
                    "4": "",
                    "5": "",
                    "6": "",
                    "7": "",
                    "8": "",
                  },
                  "name": "",
                },
                "3": Object {
                  "channels": Object {
                    "1": "",
                    "2": "",
                    "3": "",
                    "4": "",
                    "5": "",
                    "6": "",
                    "7": "",
                    "8": "",
                  },
                  "name": "",
                },
                "4": Object {
                  "channels": Object {
                    "1": "",
                    "2": "",
                    "3": "",
                    "4": "",
                    "5": "",
                    "6": "",
                    "7": "",
                    "8": "",
                  },
                  "name": "",
                },
              },
              "current_step": 1,
              "focus": Object {
                "column": 0,
                "row": 0,
              },
              "last_step": 0,
              "max_steps": 10,
              "num_plan_files": 0,
              "panel_to_show": 0,
              "plan_filename": "",
              "plan_files": Object {},
              "steps": Object {},
            }
          }
          uistatus={Object {}}
          ws_sender={[Function]}
        />
      </div>
      <div
        className="col-sm-9"
        style={
          Object {
            "display": "grid",
            "gridTemplateColumns": "1fr",
          }
        }
      >
        <div
          style={
            Object {
              "gridColumnStart": "1",
              "gridRowStart": "1",
              "padding": "10px",
            }
          }
        />
      </div>
    </div>
  </div>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={true}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "modalCenter": "styles_modalCenter__L9F2w",
        "overlay": "styles_overlay__CLSq-",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    focusTrapOptions={Object {}}
    focusTrapped={true}
    onClose={[Function]}
    open={false}
    role="dialog"
    showCloseIcon={true}
    styles={
      Object {
        "overlay": Object {
          "color": "black",
        },
      }
    }
  >
    <div
      style={
        Object {
          "margin": "20px",
        }
      }
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </Modal>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    role="alert"
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</div>
`;
